### DISCUSSION TITLE
Images not loading on Excalidraw in the site

### DISCUSSION THREAD

**Discussion by UnordinarySheep** [oleeskild/obsidian-digital-garden#640] [General]
The images are not loading why is that how can i fix it.
here is a link to the site https://digigard-one.vercel.app/7-defcon-machine-learning/drawing-2024-09-12-02-10-40-excalidraw/

![Screenshot 2024-09-25 085340](https://github.com/user-attachments/assets/41a9729a-118a-4427-bf91-8a0632c9fcca)

**Comment by UnordinarySheep** [comment]
It works with svg but not images 

i need to understand how when i load my markdown file from obsidian which is a excalidrawing but is loadable and is loading as shown in the example of the site of digital garden documentation and as i tried it. They way that the plugin works is you add dgpublish true ontop of the markdown file and it shows up in the site in the filetree sidebar . like you would see it in obsidian interface. so when i add the dh publish true to the excalidrawing it renders it also and it shows it as a drawing the problem is the files on the drwaing itself. The files that it renders on the excalidrawing  are text and eveything that you can add from the excalidraw itself but if i try to add a image onto the excalidrawing it just renders a box of where the image is plotted on the excalidrwaing canvas and the image is not loading it just shows as grey outline where it is located. how do i fix that . i tought to myself that if i find out the code that gets generated from the ecalidrawing when its rendered and converted so ican be loaded into the site is edited and the pictures get added there maybe with a link from web or some other way that it will help it load it would make it work and it wont stay just a grey outline of where they are but actually they pop up on there

**Comment by UnordinarySheep** [comment]
I tried editing the fileid that is generated in obsidian
![Screenshot 2024-09-26 220114](https://github.com/user-attachments/assets/7e15aab9-5489-4eb5-bd9f-ec0138fcfd65)


you can see the file id "6512dfc99bccdcd784a3dc901211244ad68aed93' next to the image. It was placed in this bracket that i have underlined in red. I tried changing it to a url of the image i uploaded on web but it still didnt render in the excalidrawing in the digital garden site.

![Screenshot 2024-09-26 215947](https://github.com/user-attachments/assets/ca32450a-8ebb-4465-8f82-9d95036cbd75)


SO WHAT DO I DO I NEED HELP

**Comment by sadass2** [comment]
It picks only the JSON code from the excalidrawing and the rest it doesnt. How can we make it pickup the rest of the excalidrawing.
(IF YOU OPEN THE EXCALIDRAWING AS A MARKDOWN FILE IN OBSIDIAN YOU WILL SEE THIS SCREEN!)
(WHY IS IT PICKING ONLY THE LOWEST PART OF THE MARKDOWN FILE AND HOW CAN WE MAKE IT PICK UP ALL OTHER COMPONENTS NOT ONLY THE JSON CODE OF THE DRAWING. THAT MEANS THAT IT WILL DISPLAY ONLY DRAWINGS TEXT AND WHATEVER IS A PART OF THE DRAWING WHICH IS ENCODED AS JSON CODE BUT EVERY OTHER EXTERNAL FILE LIKE IMAGES VIDEO OR GIFS WILL NOT GET PICKED BECAUSE THEY ARE NOT ENCODED AS JSON CODE BUT AS LINKS WHICH LINKS DOESNT SEEM TO WORK IN THE WEB REGARDLESS OF HOW YOU CHANGE THE CODE. 

![Screenshot 2024-09-27 183558](https://github.com/user-attachments/assets/81cd40f5-4645-402c-bc61-586c8f664809)

**Comment by UnordinarySheep** [comment]
I found what is used to generate the code for the excalidrawing and what is also used from the site 
https://github.com/zsviczian/obsidian-excalidraw-plugin/releases/tag/2.3.0

if anybody can go through it and find what we need to modify in the code so the browser picks up the images i will apreciate.
